packages/@eventual/core-runtime/src/local/clients/queue-client.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  Schedule,
+  type DurationSchedule,
+  type FifoQueue,
+  type FifoQueueHandlerMessageItem,
+  type Queue,
+  type QueueHandlerMessageItem,
+} from "@eventual/core";
+import type { QueueSendMessageOperation } from "@eventual/core/internal";
+import { ulid } from "ulidx";
+import { QueueClient } from "../../clients/queue-client.js";
+import {
+  computeScheduleDate,
+  type LocalEnvConnector,
+  type QueueProvider,
+} from "../../index.js";
+
+export interface QueueRetrieveMessagesRequest {
+  maxMessages?: number;
+  visibilityTimeout?: DurationSchedule;
+}
+
+/**
+ * TODO: implement message deduplication
```

Comment:
Issue or do? Seems straightforward?

Reply:
I guess the hard part is the content based deduplication

Reply:
The issue is the time window, but may not be hard to do. Save all of the values/hashes with a date, separated by group for fifo. Ignore messages with the same value/hash until the time is up.